### ISSUE TITLE
2.7.1 breaks custom plugin loading for local user login

### ISSUE THREAD

**Issue by robertpatrick** [rancher/dashboard#8236]
**Setup**
- Rancher version: 2.7.1
- Browser type & version: Firefox 109.0.1

**Describe the bug**

We have a custom plugin we placed in the `pkg` directory of the Rancher Dashboard source code.  When logging in with a local user, everything works as expected with 2.6.7 and 2.6.8.  Unfortunately, the plugin is not being loaded upon local user login.  If we manually force a reload, then it loads. 

**To Reproduce**

1.) Create a custom plugin in the rancher-dashboard `pkg` directory.
2.) Run `yarn dev`
3.) Log in as a local user
4.) Observe that the slider menu does not list the custom plugin.
5.) Force the browser to reload the page.
6.) Observe that the slider menu now lists the custom plugin.

**Result**
Upon local user login, users cannot see/use our plugin

**Expected Result**

Upon local user login, users can see/use our custom plugin

**Screenshots**

Slider menu immediately after login
<img width="283" alt="Screenshot 2023-02-22 at 11 24 47 AM" src="https://user-images.githubusercontent.com/31662131/220707272-48d781ac-6328-44a7-ad11-34252930bbea.png">

<img width="280" alt="Screenshot 2023-02-22 at 11 25 20 AM" src="https://user-images.githubusercontent.com/31662131/220707397-84b3d60e-f05d-4ad4-ae95-af29f73285ab.png">

**Additional context**

**Comment by richard-cox** [comment]
@robertpatrick would it be possible to validate this happens in the `master` branch?

**Comment by robertpatrick** [comment]
@richard-cox It will be painful but we will give it a shot.

**Comment by robertpatrick** [comment]
@richard-cox 

No, plugin has been in ./pkg folder since the 2.6.7 release.  Yes, it is included in the dashboard at build time.

**Comment by robertpatrick** [comment]
@richard-cox wow, that was painful--particularly because of your eslint rule changes...

Anyway, it appears to work properly in master.  Is there some workaround that I can do in 2.7.1?  If not, when can we expect the next release with this fix in it?

**Comment by aalves08** [comment]
@robertpatrick we've updated the docs for extensions (aka plugins). Some code has changed since the early versions of extensions (aka plugins).

You should check the docs regarding initialising an extension: https://rancher.github.io/dashboard/extensions/initializing-extensions which provides good information on how to register an extension. The problem might lie on that part, but it's difficult to know exactly where the problem lies. Could you try that on 2.7.1?

we also have a couple of official extensions published, which might help to check/compare with your code:
https://github.com/rancher/elemental-ui
https://github.com/kubewarden/ui

**Comment by robertpatrick** [comment]
@aalves08 We have already made one architectural change in the last year (to go from the 2.6.5 approach to the 2.6.7+ ./pkg approach.  We have been watching the new "extensions" approach but frankly, are concerned that it is not yet ready to replace what we are currently delivering and rearchitecting our plugin to pick up 2.7.1 "fixes" seems like a big ask.  Since master seems to have resolved our issue, I am tempted to just "do nothing" and wait for 2.7.2 to release.  

As we have bandwidth, we will do a POC with the new approach and see what that gets us (and what we lose).

**Comment by robertpatrick** [comment]
@aalves08 in looking through the docs on extensions, it's not clear to me that our plugin will ever fit into the currently documented extensions API but maybe I am missing something.  This extensions mechanism documentation reads more like an optional extension that the user can install (like installing a third-party plugin in an IDE).  Our approach is more of an OEM approach where we deliver the preconfigured, customized version out of the box.

**Comment by nwmac** [comment]
Can you try with the latest Rancher and re-open this issue if this is still an issue for you - this should have been fixed in the latest release.
